src/Middleware/ResponseCaching/src/ResponseCachingFeature.cs

```diff
@@ -21,10 +21,7 @@ public class ResponseCachingFeature : IResponseCachingFeature
             {
                 for (var i = 0; i < value.Length; i++)
                 {
-                    if (string.IsNullOrEmpty(value[i]))
-                    {
-                        throw new ArgumentException($"When {nameof(value)} contains more than one value, it cannot contain a null or empty value.", nameof(value));
```

Comment:
The new exception message is now one of the following:

> System.ArgumentNullException: Value cannot be null. (Parameter 'values[i]')
> System.ArgumentException: The value cannot be an empty string. (Parameter 'values[x]')

That leaves out the important detail that null or empty values are allowed if there's only one value in the array. Please revert this.

Reply:
Reverted!